pkg/cmd/repo/archive/archive.go

```diff
@@ -55,14 +68,21 @@ func archiveRun(opts *ArchiveOptions) error {
 	}
 	apiClient := api.NewClientFromHTTP(httpClient)
 
-	var toArchive ghrepo.Interface
+	repoSelector := opts.RepoArg
+	if repoSelector == "" {
+		currRepo, err := opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+		repoSelector = ghrepo.FullName(currRepo)
+	}
 
```

Comment:
I think this might be simpler:
```go
	var toArchive ghrepo.Interface

	if opts.RepoArg == "" {
		toArchive, err = opts.BaseRepo()
		if err != nil {
			return err
		}
	} else {
		repoSelector := opts.RepoArg
		if !strings.Contains(repoSelector, "/") {
			cfg, err := opts.Config()
			if err != nil {
				return err
			}

			hostname, err := cfg.DefaultHost()
			if err != nil {
				return err
			}

			currentUser, err := api.CurrentLoginName(apiClient, hostname)
			if err != nil {
				return err
			}
			repoSelector = currentUser + "/" + repoSelector
		}

		toArchive, err = ghrepo.FromFullName(repoSelector)
		if err != nil {
			return err
		}
	}
```